### PR TITLE
Add tests for error cases in did-resolver.spec.ts

### PR DESCRIPTION
**Pull request by shobitb**
This commit adds new tests in did-resolver.spec.ts to cover some of the previously uncovered error-related branching paths.

This commit contributes partly to issue #136.

Based on the output of `npm run test:node` before vs. after this commit, the `% Branch` is now increased by 9.66 percentage points, and the `% Stmt` and `% Lines` by 3.29 percentage points.

Before:
```
did-resolver.ts |   87.91 |    78.57 |   66.66 |   87.91
```

After:
```
did-resolver.ts |    91.2 |    88.23 |   66.66 |    91.2 |
```

### CODE REVIEW

**Review by thehenrytsai — COMMENTED** [review]
@shobitb thanks for helping us increase CC! Could you opportunistically convert the `Error` used in DidResolver into `DwnError` and introduce new error codes?  Search for `DwnError` and you'll find pattern used in code and tests.

**Comment by shobitb** [comment]
> @shobitb thanks for helping us increase CC! Could you opportunistically convert the `Error` used in DidResolver into `DwnError` and introduce new error codes? Search for `DwnError` and you'll find pattern used in code and tests.

Yes, most def. Let me take a look and will send an updated PR.

File: src/did/did.ts [hunk 1]
```diff
@@ -16,13 +18,13 @@ export class Did {
    */
   public static validate(did: unknown): void {
     if (typeof did !== 'string') {
-      throw new Error(`DID is not string: ${did}`);
+      throw new DwnError(DwnErrorCode.DidNotValid, `DID is not string: ${did}`);
```
Comment (thehenrytsai) [inline src/did/did.ts:21]: Nit: As a coding guideline, we never reuse the same error code. So introduce a new one, perhaps:

```suggestion
      throw new DwnError(DwnErrorCode.DidNotString, `DID is not string: ${did}`);
```

**Review by thehenrytsai — COMMENTED** [review]
Last nit then good to go!

**Comment by shobitb** [comment]
> Last nit then good to go!

Gotcha!  I originally thought the codes were like HTTP codes, where a generic code could be used with different specific error messages.

Added a new code for `DidNotString` in the latest commit

**Review by thehenrytsai — APPROVED** [review]
Good to merge! Thanks again!!

**Pull request merged**